sed/binning/binning.py

```diff
@@ -43,34 +44,35 @@ def bin_partition(
     """Compute the n-dimensional histogram of a single dataframe partition.
 
     Args:
-        part: dataframe on which to perform the histogram.
-            Usually a partition of a dask DataFrame.
-        bins: Definition of the bins. Can  be any of the following cases:
-            - an integer describing the number of bins in on all dimensions
-            - a tuple of 3 numbers describing start, end and step of the binning range
-            - a np.arrays defining the binning edges
-            - a list (NOT a tuple) of any of the above (int, tuple or np.ndarray)
-            - a dictionary made of the axes as keys and any of the above as values.
-            This takes priority over the axes and range arguments.
-        axes: The names of the axes (columns) on which to calculate the histogram.
-            The order will be the order of the dimensions in the resulting array.
-        ranges: list of tuples containing the start and end point of the binning range.
-        histMode: Histogram calculation method. Choose between
-            "numpy" which uses numpy.histogramdd, and "numba" which uses a
-            numba powered similar method.
-        jitter: a list of the axes on which to apply jittering.
-            To specify the jitter amplitude or method (normal or uniform noise)
-            a dictionary can be passed.
-            This should look like jitter={'axis':{'amplitude':0.5,'mode':'uniform'}}.
+        part (Union[dask.dataframe.core.DataFrame, pd.DataFrame]): dataframe on which
```

Comment:
Does autodoc need to have all types explicitly mentioned in the comment?
It seems dangerous to leave to the programmer to change the data type comment if the type hint is changed.

Reply:
This is the difference of the two versions:
With typing in docstring:
![image](https://user-images.githubusercontent.com/53396064/223083353-6f28d9ea-534c-4636-bde4-d23a96104e80.png)
Without typing in docstring:
![image](https://user-images.githubusercontent.com/53396064/223083952-b9aeb7a5-1f52-44df-a42f-6ebecbd8eb69.png)

We can change that for sure again, if you think the latter is better.

Reply:
You have a good point. Would be so nice to use the python 3.10 syntax for type hints instead, but it unfortunately breaks retro-compatibility. 
Do you think there might be a git workflow step which checks that type hints match the comments?

Reply:
What happens if there is a mix of both cases? As, for example, if we do not update the docstring of a function. Will sphinx add it where it is missing? or will it add it everywhere if we toggle it on?

Reply:
If they are missing, they are filled in as shown in the example. I don't know what happens if they are contradicting, I suppose the ones in the docstring win...

Reply:
That would be perfect then. We can modify the ones we notice, and the others will have a less readable, yet complete, documentation!